pkg/controllers/bgp/bgpconf_controller.go

```diff
@@ -190,9 +246,28 @@ func (r *BgpConfReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	// The policy field must be indexed by the manager, so that we will be able to lookup BgpConf by a referenced ConfigMap name.
+	err := mgr.GetFieldIndexer().
+		IndexField(context.Background(), &v1alpha2.BgpConf{}, policyField, func(rawObj runtime.Object) []string {
+			// Extract the ConfigMap name from the BgpConf Spec, if one is provided
+			bgpConf := rawObj.(*v1alpha2.BgpConf)
+			if bgpConf.Spec.Policy == "" {
+				return nil
+			}
+			return []string{bgpConf.Spec.Policy}
+		})
+	if err != nil {
+		return err
+	}
+
 	ctl, err := ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha2.BgpConf{}).
 		WithEventFilter(p).
```

Comment:
``` diff
-               For(&v1alpha2.BgpConf{}).
-               WithEventFilter(p).
+               For(&v1alpha2.BgpConf{}, builder.WithPredicates(p)).
+               //WithEventFilter(p).
```

Reply:
`WithEventFilter` is a global filter. It will filters out the configmap. This is not what we expected

Reply:
Hmm, instead of removing/commenting the `EventFilter` entirely, what do you think about adding a check for a `ConfigMap` in `shouldReconcile`?

Reply:
> check for a ConfigMap in shouldReconcile?

high coupling